wellcomecollection/catalogue-pipeline
# Better relation embedder cli

## What does this change?

Previously, CLIMain was using Try.get, meaning that it would throw if any of the lines could not be transformed into a Batch.  This was not a problem, but it's a bit neater and more idiomatic to allow valid batches to work when other batches are invalid.

## How to test

Add a naughty line to the start middle of  ` pipeline/relation_embedder/relation_embedder/scripts/batches.txt `, then 

`cat pipeline/relation_embedder/relation_embedder/scripts/batches.txt | AWS_PROFILE=platform-developer python pipeline/relation_embedder/relation_embedder/scripts/cli.py 2024-11-18`

The good lines should work as expected, the bad one should log an ERROR.

## How can we measure success?

Part of the reason for this change is that I think some of this pattern might be able to be reused in the Real Life application.  Currently, this path is only used by developers doing som kind of local investigation, but if/when we reduce our dependence on Pekko, we might want to slightly rewrite this trait and use it to read from SQS Lambda input.

## Have we considered potential risks?

Aborting the whole thing when faced with invalid data does make it more obvious when things go wrong, but I think the normal use case for this will be to run a small number of batches at any one time, and the logging output is clear when an error has occurred, so I think this is better than it was.
